psl/psl-core/src/validate/validation_pipeline/validations.rs

```diff
@@ -124,6 +125,10 @@ pub(super) fn validate(ctx: &mut Context<'_>) {
                 fields::validate_length_used_with_correct_types(field_attribute, attribute, ctx);
             }
         }
+
+        for field in model.relation_fields() {
+            relation_indexes::validate_missing_relation_indexes(model, field, ctx);
+        }
```

Comment:
sorry, I missed that previously but that should move below line 100, we don't want to iterate over relation fields twice

Reply:
I agree that iterating over relation fields twice is suboptimal, I wonder if it makes sense to run the `validate_missing_relation_indexes` check before the other index validations are executed though. If that's safe, I can happily move this after line 100 :)

Reply:
yep, the order doesn't really matter, we never return early.

Reply:
Done